expression/builtin_other_vec_generated.go

```diff
@@ -0,0 +1,414 @@
+
+	args0 := buf0.Int64s()
+	result.ResizeInt64(n, true)
+	r64s := result.Int64s()
+	hasNull := make([]bool, n)
+	isUnsigned0 := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	var compareResult int
+
+	for j := 1; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalInt(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		isUnsigned := mysql.HasUnsignedFlag(b.args[j].GetType().Flag)
+		args1 := buf1.Int64s()
+		buf1.MergeNulls(buf0)
+		for i := 0; i < n; i++ {
+			if buf1.IsNull(i) {
+				hasNull[i] = true
+				continue
+			}
+			arg0 := args0[i]
+			arg1 := args1[i]
+			compareResult = 1
+			switch {
+			case (isUnsigned0 && isUnsigned), (!isUnsigned0 && !isUnsigned):
+				if arg1 == arg0 {
+					compareResult = 0
+				}
+			case !isUnsigned0 && isUnsigned:
+				if arg0 >= 0 && arg1 == arg0 {
+					compareResult = 0
+				}
+			case isUnsigned0 && !isUnsigned:
+				if arg1 >= 0 && arg1 == arg0 {
+					compareResult = 0
+				}
+			}
+			if compareResult == 0 {
+				result.SetNull(i, false)
+				r64s[i] = 1
+			}
+		} // for i
+	} // for j
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			result.SetNull(i, hasNull[i])
+		}
+	}
+	return nil
+}
+
+func (b *builtinInIntSig) vectorized() bool {
+	return true
+}
+
+func (b *builtinInStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	result.ResizeInt64(n, true)
+	r64s := result.Int64s()
+	hasNull := make([]bool, n)
+	var compareResult int
+
+	for j := 1; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalString(b.ctx, input, buf1); err != nil {
+			return err
+		}
+		for i := 0; i < n; i++ {
+			if buf1.IsNull(i) || buf0.IsNull(i) {
+				hasNull[i] = true
+				continue
+			}
+			arg0 := buf0.GetString(i)
+			arg1 := buf1.GetString(i)
+			compareResult = types.CompareString(arg0, arg1)
+			if compareResult == 0 {
+				result.SetNull(i, false)
+				r64s[i] = 1
+			}
+		} // for i
+	} // for j
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			result.SetNull(i, hasNull[i])
+		}
+	}
+	return nil
+}
+
+func (b *builtinInStringSig) vectorized() bool {
+	return true
+}
+
+func (b *builtinInDecimalSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	args0 := buf0.Decimals()
+	result.ResizeInt64(n, true)
+	r64s := result.Int64s()
```

Comment:
We have to reset this array(set all elements of it to zero). @mmyj

Reply:
I will fix it.